### DISCUSSION TITLE
[Help Needed] How to integrate custom EMF Meta Model into Sirius Web

### DISCUSSION THREAD

**Discussion by LeoWPereira** [eclipse-sirius/sirius-components#3910] [Q&A]
Hello everyone.
For a project I have a EMF meta model created using ObeoDesigner . Now I'm aiming to integrate that one into Sirius Web, but I'm facing issues doing so.

Based on documentation, what I've tried to do so far is add the `@Bean` configuration, like the following:
![image](https://github.com/user-attachments/assets/087a3150-00a0-4937-8894-368968647398)

The maven build succeeds and the package itself is built (I do see the `target` folder being created). 

When I run the fat jar though, the meta model is not visible and when uploading a model instance I only see:
![image](https://github.com/user-attachments/assets/d16fef46-1214-4f7b-971a-e6199f83b2b6)

Trying to debug the added `@Bean` seem like they are not being executed when I start the Spring application.

Did I miss anything obvious here? Any help or tip?

**Comment by pcdavid** [comment]
Hi,

By default the Spring configuration defined in `org.eclipse.sirius.web.starter.SiriusWebStarterConfiguration` will tell Spring Boot to only scan for components [in specific Java packages](https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/backend/sirius-web-starter/src/main/java/org/eclipse/sirius/web/starter/SiriusWebStarterConfiguration.java#L51).

If you defined your `@Bean` in a different Java package it will not be found by Spring Boot.

The correct way is to define your own `@SpringBootApplication` in the same way as `sirius-web-application` and provide a custom `@ComponentScan` definition. See for example how SysON defines its entry point application [here](https://github.com/eclipse-syson/syson/blob/main/backend/application/syson-application/src/main/java/org/eclipse/syson/SysONApplication.java) in a way that makes Spring Boot also scan for its own SysON-specific components in `"org.eclipse.syson"` and sub-packages.
